packages/astro/src/content/utils.ts

```diff
@@ -50,11 +51,19 @@ export const msg = {
 };
 
 export function extractFrontmatterAssets(data: Record<string, any>): string[] {
-	const assets = Object.values(data).filter(
-		(value) => typeof value === 'object' && value['__astro_asset'] == true
-	);
+	function findAssets(potentialAssets: Record<string, any>): ImageMetadata[] {
+		return Object.values(potentialAssets).map((entry: any) => {
+			if (typeof entry === 'object') {
+				if (entry.__astro === true) {
+					return entry;
+				} else {
+					return findAssets(entry);
+				}
+			}
```

Comment:
1. if `typeof entry === 'object'` is `false`, we don't map to anything, and we return `undefined`, is that correct?
2. Since we are using recursion with `findAssets`, maybe this is a good situation to use [`flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)

Reply:
Yeah, logic is a bit faulty, as tests have shown... Will fix

How would you use `flatMap` in this context? It only flatten a single level

Reply:
If we know that we can only have one level, maybe we don't need to use a recursive approach. At first I thought we could have many levels.

Reply:
The images can be infinitely deep, as Markdown frontmatters can include nested elements

Reply:
You could `.flat(Infinity)`. `.flat` accepts a number that tells the function how many levels deep the flattening needs to occur.